contracts/libraries/VaultLifecycleTreasury.sol

```diff
@@ -766,4 +786,158 @@ library VaultLifecycleTreasury {
         }
         return friday8am;
     }
+
+    function getNextExpiry(
+        uint256 currentExpiry, 
+        uint256 day, 
+        uint256 period, // no of days in between option selling
+        bool initial
+    )
+        internal
+        pure
+        returns (uint256 nextExpiry)
+    {   
+        if (initial) {
+            uint256 weekday = getWeekday(currentExpiry) == 0
+                ? 7
+                : getWeekday(currentExpiry);
+
+            nextExpiry = (weekday >= day
+                ? currentExpiry - (weekday - day) * 1 days + 1 weeks
+                : currentExpiry + (day - weekday) * 1 days);
+        } else if (period == 30) {
+            nextExpiry = getNextMonthExpiry(currentExpiry, day); 
+        } else if (period % 7 == 0) {
+            uint256 weekday = getWeekday(currentExpiry) == 0
+                ? 7
+                : getWeekday(currentExpiry);
+
+            nextExpiry = (weekday >= day
+                    ? currentExpiry - (weekday - day) * 1 days
+                    : currentExpiry + (day - weekday) * 1 days)
+                + (period / 7) * 1 weeks;
+        }
+
+        nextExpiry = nextExpiry - (nextExpiry % (24 hours)) + (8 hours);
+        
+        return nextExpiry;
+    }
+
+    /************************************************
+     *  DATE HELPERS
```

Comment:
I'm a bit worried the date helpers could be buggy because dealing with date is a nightmare.  We really need to audit this and write tests, compare the date functions to another library, maybe in a different MR.

Reply:
Yes I agree, I'm actually referencing from here https://github.com/pipermerriam/ethereum-datetime/blob/master/contracts/DateTime.sol

Reply:
Maybe we should import this as a library?